miniapps/dpg/diffusion.cpp

```diff
@@ -0,0 +1,410 @@
+//                       MFEM Ultraweak DPG example for diffusion
+//
+// Compile with: make diffusion
+//
+// sample runs
+//   diffusion -m ../../data/star.mesh -o 3 -ref 1 -do 1 -prob 1 -sc
+//   diffusion -m ../../data/inline-tri.mesh -o 2 -ref 2 -do 1 -prob 0
+//   diffusion -m ../../data/inline-quad.mesh -o 4 -ref 1 -do 2 -prob 0 -sc
+//   diffusion -m ../../data/inline-tet.mesh -o 3 -ref 0 -do 1 -prob 1 -sc
+
+// Description:
+// This example code demonstrates the use of MFEM to define and solve
+// the "ultraweak" (UW) DPG formulation for the Poisson problem
+
+//       - Δ u = f,   in Ω
+//           u = u₀, on ∂Ω
+
+// It solves two kinds of problems
+// a) f = 1 and u₀  = 0 (like ex1)
+// b) A manufactured solution problem where u_exact = sin(π * (x + y + z)).
+//    This example computes and prints out convergence rates for the L2 error.
+
+// The DPG UW deals with the First Order System
+//   ∇ u - σ = 0, in Ω
+// - ∇⋅σ     = f, in Ω
+//        u  = u₀, in ∂Ω
+
```

Comment:
In each example, it would be helpful to add a reference to the DPG literature where the reader can learn more about the DPG formulation.